modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -1996,114 +1892,68 @@ public int getMaxBackBufferMemory(int width, int height) {
      * @throws IOException
      * @throws IllegalFilterException
      */
-    private void processStylers(final Graphics2D graphics,
-            final Layer layer, AffineTransform at,
-            CoordinateReferenceSystem destinationCrs, Envelope mapArea,
-            Rectangle screenSize, String layerId) throws Exception {
-        /*
-         * DJB: changed this a wee bit so that it now does the layer query AFTER
-         * it has evaluated the rules for scale inclusion. This makes it so that
-         * geometry columns (and other columns) will not be queried unless they
-         * are actually going to be required. see geos-469
-         */
+    private void processStylers(final Graphics2D graphics, final Layer layer, String layerId)
+            throws Exception {
         // /////////////////////////////////////////////////////////////////////
         //
         // Preparing feature information and styles
         //
         // /////////////////////////////////////////////////////////////////////
-        final MapLayer currLayer = new MapLayer(layer);        
-        final Style style = currLayer.getStyle();
-        final FeatureSource featureSource = currLayer.getFeatureSource();
-
-        final CoordinateReferenceSystem sourceCrs;
-        final NumberRange scaleRange = NumberRange.create(scaleDenominator,scaleDenominator);
-        final ArrayList<LiteFeatureTypeStyle> lfts ;
-
-        if ( featureSource != null ) {
-
-            final FeatureType schema = featureSource.getSchema();
+        final FeatureSource featureSource = layer.getFeatureSource();
+        if (featureSource == null) {
+            throw new IllegalArgumentException("The layer does not contain a feature source");
```

Comment:
The layerId is just a counter, used to generate unique ids for the LabelCache. Going to use the layer title instead.